### PR TITLE
Remove async from renamer

### PR DESCRIPTION
**Pull request by davkean**
This includes https://github.com/dotnet/project-system/pull/5433, so only review https://github.com/dotnet/project-system/commit/6475aa759c8688fc71e8550b28542850ab231707.

These code paths were async because they used IVsService<T>, moved to IVsUIService as they are using only UI-bound services.

### CODE REVIEW

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Refactor/VisualStudioRefactorNotifyService.cs [hunk 1]
```diff
@@ -11,50 +12,124 @@
...
-        public Task DeactivateAsync() => UnloadAsync();
+        public void OnBeforeGlobalSymbolRenamed(string projectPath, IEnumerable<string> filePaths, string rqName, string newName)
+        {
+            IVsHierarchy? projectHierarchy = GetProjectHierarchy(projectPath);
+            if (projectHierarchy == null)
+            {
+                return;
+            }
+
+            if (!(projectHierarchy is IVsHierarchyRefactorNotify refactorNotify))
```
Comment (jmarolf) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Refactor/VisualStudioRefactorNotifyService.cs]: couldn't this just be a single if with an `||`?

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Refactor/VisualStudioRefactorNotifyService.cs [hunk 2]
```diff
@@ -11,50 +12,124 @@
...
+
+        public void OnAfterGlobalSymbolRenamed(string projectPath, IEnumerable<string> filePaths, string rqName, string newName)
+        {
+            IVsHierarchy? projectHierarchy = GetProjectHierarchy(projectPath);
+            if (projectHierarchy == null)
+            {
+                return;
+            }
+
+            if (!(projectHierarchy is IVsHierarchyRefactorNotify refactorNotify))
```
Comment (jmarolf) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Refactor/VisualStudioRefactorNotifyService.cs]: same question

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Refactor/VisualStudioRefactorNotifyService.cs (hunk 1 above)
Comment (davkean) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Refactor/VisualStudioRefactorNotifyService.cs]: I didn't change this code.

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Refactor/VisualStudioRefactorNotifyService.cs (hunk 2 above)
Comment (davkean) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Refactor/VisualStudioRefactorNotifyService.cs]: I didn';t change this code.

**Pull request merged**